python/taichi/lang/__init__.py

```diff
@@ -398,6 +445,11 @@ def no_activate(*args):
 
 
 def block_local(*args):
+    """Hints Taichi to cache the list of fields into the shared memory.
```

Comment:
Do we also want to mention that it can potentially bring some speedup but it's not guaranteed?

Reply:
Added the link to the doc explaining BLS